lib/console.js

```diff
@@ -49,9 +55,32 @@ Console.prototype.error = Console.prototype.warn;
 
 
 Console.prototype.dir = function(object, options) {
-  this._stdout.write(util.inspect(object, util._extend({
-    customInspect: false
-  }, options)) + '\n');
+  this._stdout.write((this._group ? util.inspect(object, util._extend({
+        customInspect: false }, options)).replace(/^/gm, this._prefix) :
+          util.inspect(object, util._extend({ customInspect: false },
+              options)).replace(/^/gm, this._prefix)) + '\n');
+};
+
+
+Console.prototype.group = function() {
+  if (arguments.length) {
+    this.log.apply(this, arguments);
+  }
+  this._group++;
+  this._prefix = '  '.repeat(this._group);
+};
+
+
+Console.prototype.groupCollapsed = function() {
+  this.group.apply(this, arguments);
+};
+
+
+Console.prototype.groupEnd = function() {
+  if ( this._group > 0 ) {
+    this._group--;
+    this._prefix = (this._group ? '  '.repeat(this._group) : '');
```

Comment:
I think this is right place for this line:
```js
if (_group` == 0) //do nothing
else //modify something.
```
And about:
```js
this._prefix = (this._group ? '  '.repeat(this._group) : '');
```
generate some levels of indentation
or
```js
this._prefix = this._prefix.slice(0, -2);
```
trim two characters from indentation level